src/lib/components/legend/Legend.svelte

```diff
@@ -42,167 +33,96 @@
 		{ value: 'fundingSource', label: 'Funding' }
 	];
 
-	function handleModeChange(event: Event) {
-		const target = event.target as HTMLInputElement;
-		visualState.update(state => ({ 
-			...state, 
-			colorMode: target.value as typeof state.colorMode,
+	function handleModeChange(event: Event & { currentTarget: HTMLInputElement }) {
+		visualState.set({
+			mode: event.currentTarget.value as Mode,
 			filters: new Set()
-		}));
+		});
 	}
 
 	function toggleFilter(label: string) {
-		const newFilters = new Set($visualState.filters);
-		
-		if (newFilters.has(label)) {
-			newFilters.delete(label);
-		} else {
-			if (newFilters.size === 0) {
-				newFilters.add(label);
-			} else {
-				if (label === 'Other') {
-					newFilters.add('Other');
-				} else if (CATEGORIES[$visualState.colorMode].includes(label)) {
-					newFilters.delete('Other');
-					newFilters.add(label);
-				} else {
-					newFilters.add(label);
-				}
-			}
+		const updatedFilters = new Set($visualState.filters);
+
+		if (updatedFilters.has(label)) {
+			updatedFilters.delete(label);
+		} else if (label === 'Other') {
+			updatedFilters.add(label);
+		} else if (CATEGORIES[$visualState.mode].includes(label)) {
+			updatedFilters.delete('Other');
+			updatedFilters.add(label);
 		}
-		
-		const updatedFilters = new Set(newFilters);
-		visualState.update(state => ({
-			...state,
-			filters: updatedFilters
-		}));
-	}
 
-	$: isItemActive = (label: string) => $visualState.filters.has(label);
```

Comment:
Here you're actually declaring a reactive _function_. In general, you never really want to do this; this means the function has to be reallocated every time the `$visualState` store changes. Instead, prefer just inlining these kinds of expressions directly where they're used (for example, I inlined this on L119 of the changes).